.github/workflows/notify-fork-prs.yml

```diff
@@ -0,0 +1,46 @@
+---
+name: "Notify on PRs from forks"
+
+on:  # yamllint disable-line rule:truthy
+  pull_request_target:
+    branches:
+      - main
+    types:
+      - opened
+
+jobs:
+  notify-fork-prs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Notify on PRs from forks
+        id: hash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          BASE_REPO: ${{ github.event.pull_request.base.repo.full_name }}
+          HEAD_REPO: ${{ github.event.pull_request.head.repo.full_name }}
+          USER: ${{ github.actor }}
+        run: |
+          if [ "$BASE_REPO" = "$HEAD_REPO" ]; then
+            echo "PR is from a branch on this repo."
+            exit
+          fi
+
+          echo "PR is from $HEAD_REPO to $BASE_REPO by $USER."
+
+          # Check if the user has write access to the repository.
+          check_status="$(curl -sL \
+            -o /dev/null \
+            -w '%{http_code}' \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/$BASE_REPO/collaborators/$USER")"
```

Comment:
FWIW, you can just use `gh api` in actions. Which I tend to think is a little easier to read. :)